authorization-common/src/main/java/com/sequenceiq/authorization/utils/EventAuthorizationUtils.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.authorization.utils;
+
+import static com.sequenceiq.cloudbreak.util.NullUtil.throwIfNull;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.resource.AlternativeAuthorizationResourceActionProvider;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.service.CommonPermissionCheckingUtils;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+
+@Component
+public class EventAuthorizationUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventAuthorizationUtils.class);
+
+    private AlternativeAuthorizationResourceActionProvider actionAlternativeProvider;
+
+    private CommonPermissionCheckingUtils permissionCheckingUtils;
+
+    public EventAuthorizationUtils(CommonPermissionCheckingUtils permissionCheckingUtils,
+            AlternativeAuthorizationResourceActionProvider actionAlternativeProvider) {
+        this.actionAlternativeProvider = actionAlternativeProvider;
+        this.permissionCheckingUtils = permissionCheckingUtils;
+    }
+
+    public void checkPermissionBasedOnResourceTypeAndCrn(Collection<EventAuthorizationDto> eventAuthorizationDtos) {
+        throwIfNull(eventAuthorizationDtos,
+                () -> new IllegalArgumentException("The collection of " + EventAuthorizationDto.class.getSimpleName() + "s should not be null!"));
+        for (EventAuthorizationDto dto : eventAuthorizationDtos) {
+            String resourceType = dto.getResourceType();
+            Arrays.asList(AuthorizationResourceAction.values()).stream()
+                    .filter(action -> isResourceTypeHasDescribeOrGetAction(action, resourceType))
+                    .findFirst()
+                    .ifPresentOrElse(
+                            action -> checkPermissionForResource(action, dto.getResourceCrn()),
```

Comment:
So basically we check permission for every event. This can cause some problems since UMS is implementing rate limiting. If most of the events are about one resource than it will be cached after the first call but if those cover lot of different crns than we can put a lot of load on UMS and actually they will limit us.